src/client/QXmppEntityTimeManager.h

```diff
@@ -26,28 +26,37 @@
 
 #include "QXmppClientExtension.h"
 
+#include <variant>
+
+template<class T>
+class QFuture;
 class QXmppEntityTimeIq;
 
+///
 /// \brief The QXmppEntityTimeManager class provided the functionality to get
 /// the local time of an entity as defined by \xep{0202}: Entity Time.
 ///
 /// \ingroup Managers
-
+///
 class QXMPP_EXPORT QXmppEntityTimeManager : public QXmppClientExtension
 {
     Q_OBJECT
 
 public:
-    QString requestTime(const QString& jid);
+    QString requestTime(const QString &jid);
+
+    using EntityTimeResult = std::variant<QXmppEntityTimeIq, QXmpp::PacketState>;
```

Comment:
should probably include `QXmppStanza::Error` for consistency and so the library user doesn't need to do the check.